src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.addToAlive(this);
+  }
+
+  static addToAlive(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeAlive(animal) {
+    Animal.alive = Animal.alive.filter(aliveAnimal => aliveAnimal !== animal);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    }
+
+    this.hidden = true;
+  }
```

Comment:
seems like toggle, but more difficult, this method should not toggle value, only hide